woocommerce/woocommerce-ios
# Swap the order of prologue CTAs as an A/B experiment

Closes: #7550 


### Description


Please see pbxNRc-1VA-p2 for more details about the A/B experiment. Basically, we want to run an A/B experiment on the order of two buttons on the login prologue screen:

- “Enter Your Store Address”: primary in `control`, secondary in `treatment`
- “Continue with WordPress.com”: secondary in `control`, primary in `treatment`

The challenging part of the implementation is from this UI being the initial UI shortly after the app launch, while the A/B test setup and fetching assignments are async. I thought and tried a few different things, we could delay the app launch to wait for the A/B test setup completion but that doesn't sound ideal. Eventually, I decided to take advantage of the current onboarding screen - while a new user is still interacting with the onboarding screen, the A/B test setup is most likely complete. When the onboarding is shown, we only configure and show the authenticator when dismissing the onboarding screen.

### Testing instructions


#### `control` variation

This is the default variation, and most likely the one that is shown before the experiment is launched.

- Delete the app
- Launch the app from the PR branch --> the onboarding screen should be shown, and an event is logged `🔵 Tracked application_installed, properties: [AnyHashable("after_abtest_setup"): true, AnyHashable("prologue_experiment_variant"): "control"]`
- Finish or dismiss the onboarding --> the prologue screen should be shown, and `Enter Your Store Address` should be the primary CTA like before. in Xcode console, an event should be logged `🔵 Tracked login_accessed, properties: [AnyHashable("prologue_experiment_variant"): "control"]`

#### Testing the `treatment` variation

Unfortunately, it requires the knowledge of the anonymous ID which we don't know until after the app launch. Because we don't delay the authenticator setup & UI if the onboarding isn't shown, relaunching the app is required for the new variation to take effect.

Prerequisites:
- Please follow the setup in PCYsg-Fq7-p2#assignments-api for proxied sandboxed API requests: connect to sandbox via ssh, and point WP.com API to the sandbox IP address (`ifconfig`) in the local matchine's `/etc/hosts`
- In Xcode, set a breakpoint at `ExPlatService` L47 where `anon_id` is set in the URL request
- Make an API request `PATCH /wpcom/v2/experiments/0.1.0/assignments` with the following `body` and make sure it's successful:
```
{
    "variations": {
        "woocommerceios_login_prologue_button_order": "treatment"
    },
    "anon_id": "anon_id_from_the_previous_step",
    "username_override": ""
}
```
- Launch the app, and let it sit for a bit
- Relaunch the app --> the treatment variation should be shown, and `Continue with WordPress.com` should become the primary CTA. in Xcode console, an event should be logged `🔵 Tracked login_accessed, properties: [AnyHashable("prologue_experiment_variant"): "treatment"]`

#### Offline case when A/B test does not complete in time

- [ ] @jaclync tests this case

- Turn the device to offline mode
- Delete the app
- Install the app --> the app should work normally
- Relaunch the app --> the app should work normally

### Screenshots



---
